sklearn/linear_model/base.py

```diff
@@ -49,7 +49,8 @@ def make_dataset(X, y, sample_weight, random_state=None):
     """Create ``Dataset`` abstraction for sparse and dense inputs.
 
     This also returns the ``intercept_decay`` which is different
-    for sparse datasets.
+    for sparse datasets. ``intercept_decay`` is not used for
```

Comment:
Why? Wouldn't we have the same offset oscillation issues when the data is sparse?

Also, I am not sure this is right place to put this comment. I would move it as a comment before the call to `make_dataset` in `SGDOneClassSVM._fit`

Reply:
I moved the comment to `_fit_oneclass` before the call to `make_dataset`.

For the moment I have preferred to allow the user to set this value in the `fit` method using the `offset_decay` parameter and not use 0.01 by default for sparse data (and 1 for dense data). The default value of the `offset_decay` is set to 1  and this is the value used in the benchmark and in the example but more experiments would be needed.

Besides I don't really understand this condition for sparse data only. I understand that we might want to have a different learning rate for the intercept as explained in the Stochastic Gradient Descent Tricks by Leon Bottou and as done in the code available at [his sgd page](http://leon.bottou.org/projects/sgd). However it seems that enforcing a smaller learning rate for the intercept is done for any kind of data and not only for sparse data:

````cpp
/// Perform one iteration of the SGD algorithm with specified gains
void
SvmSgd::trainOne(const SVector &x, double y, double eta)
{
  double s = dot(w,x) / wDivisor + wBias;
  // update for regularization term
  wDivisor = wDivisor / (1 - eta * lambda);
  if (wDivisor > 1e5) renorm();
  // update for loss term
  double d = LOSS::dloss(s, y);
  if (d != 0)
    w.add(x, eta * d * wDivisor);
  // same for the bias
#if BIAS
  double etab = eta * 0.01;
#if REGULARIZED_BIAS
  wBias *= (1 - etab * lambda);
#endif
  wBias += etab * d;
#endif
}
````